### PR TITLE
zed query: support HTTP, HTTPS URLs as input files

### LINKED ISSUE #734: Support http/https locations in zq
Add support for http/https-located data to zq, similar to the s3 support added in #704 .

### PR DESCRIPTION
**Pull request by nwt**
This makes the following command work.

    zq -z https://github.com/brimdata/zed/raw/main/testdata/babble.zson

Closes #734.

### CODE REVIEW

**Comment by mccanne** [comment]
I thought you were going to unify proc.DataAdaptor.Get and Open...?

File: cmd/zed/query/command.go [hunk 1]
```diff
@@ -211,6 +211,19 @@ func ParseSourcesAndInputs(paths, includes Includes) ([]string, ast.Proc, error)
 	return paths, query, nil
 }
 
+func isURLWithKnownScheme(path string, schemes ...string) bool {
```
Comment (mccanne) [inline cmd/zed/query/command.go:214]: Doesn't this belong in pkg/storage then inputflags would call it?

**Review by mccanne — APPROVED** [review]

**Comment by nwt** [comment]
> I thought you were going to unify proc.DataAdaptor.Get and Open...?

I am, but in a separate PR.

File: cmd/zed/query/command.go (hunk 1 above)
Comment (nwt) [reply cmd/zed/query/command.go:214]: Let's do this later.

**Pull request merged**